logstash-core/lib/logstash/runner.rb

```diff
@@ -148,25 +147,29 @@ def run(args)
     begin
       LogStash::SETTINGS.from_yaml(LogStash::SETTINGS.get("path.settings"))
     rescue => e
-      @logger.subscribe(STDOUT)
-      @logger.warn("Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options")
-      @logger.fatal("Failed to load settings file from \"path.settings\". Aborting...", "path.settings" => LogStash::SETTINGS.get("path.settings"), "exception" => e.class, "message" => e.message)
+      logger.warn("Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options")
+      logger.fatal("Failed to load settings file from \"path.settings\". Aborting...", "path.settings" => LogStash::SETTINGS.get("path.settings"), "exception" => e.class, "message" => e.message)
       exit(-1)
     end
 
+    # Configure Logstash logging facility, this need to be done before everything else to
+    # make sure the logger has the correct settings and the log level is correctly defined.
+    # TODO(talevy): cleanly support `path.log` setting in log4j
+    log4j_config_location = setting("path.settings") + "/log4j2.properties"
+    LogStash::Logging::Logger::initialize(log4j_config_location)
```

Comment:
I think it would be great to surround these 2 lines with a `if` test that checks if `log4j.configurationFile` system property is not present (using `System.getProperty` method).
With this simple code, it enables 2 interesting things :
- logstash user can define a custom log4j config location
- logstash user can choose another log4j config format : xml, yaml or json.